### PR TITLE
[DUOS-723][risk=no] Associate Consent to Dataset

### PR DESCRIPTION
**Pull request by lindojong**
Addresses: https://broadinstitute.atlassian.net/browse/DUOS-723
Foundational work to create a consent from a given dataset.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've followed [the instructions](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/DataBiosphere/consent/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing
- [ ] I've updated Swagger to reflect any API changes

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 1]
```diff
@@ -1,29 +1,91 @@
...
+            String associationType = AssociationType.SAMPLESET.getValue();
+            consentDAO.insertConsentAssociation(consentId, associationType, dataset.getDacId());
+            return consentDAO.findConsentById(consentId);
+        } else {
+            throw new NullPointerException("Dataset is missing Data Use information. Consent could not be created.");
+        }
+
+    }
+
+    private boolean isConsentDataUseManualReview(DataUse dataUse) {
```
Comment (lindojong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java:77]: I was having a bit of trouble understanding what to do with this, as it appears that many of the fields in DataUse won't actually have values, so it wouldn't be safe to arbitrarily access them. In v2, we can "fix" this by guaranteeing that all these fields are defaulted to false on the UI on the info coming in, but that would be a departure from how things are done today.

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java [hunk 2]
```diff
@@ -2,32 +2,93 @@
...
+             * translated use restriction
+             */
+            consentDAO.insertConsent(consentId, manualReview, null, dataset.getDataUse().toString(),
+                  null, name, null, createDate, createDate, null,
+                  true, groupName, dataset.getDacId());
+            String associationType = AssociationType.SAMPLESET.getValue();
+            consentDAO.insertConsentAssociation(consentId, associationType, dataset.getDacId());
+            return consentDAO.findConsentById(consentId);
+        } else {
+            throw new NullPointerException("Dataset is missing Data Use information. Consent could not be created.");
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/DatasetService.java]: This should be an `IllegalArgumentException`

File: src/main/java/org/broadinstitute/consent/http/service/DatasetService.java (hunk 1 above)
Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/service/DatasetService.java:77]: I think that's fine. The code checks for null-ness so we shouldn't fail in the case you're describing. "Manual Review"-ness is purely a function of checking a variety of fields for any value at all.

**Review by rushtong — APPROVED** [review]
One minor codacy nit inline, but otherwise, 👍

**Review by rushtong — APPROVED** [review]

**Pull request merged**